### PR TITLE
Allow multiple dids per entity

### PR DESCRIPTION
**Pull request by jakubkoci**

### CODE REVIEW

File: src/context.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { createSchemaService } from './schema/service'
+import { createEntityRepository } from './entity/mongoRepository'
+import { createEntityService } from './entity/service'
+
+interface IO {
+  database: Db
+}
+
+export async function createAppContext(io: IO) {
+  const { database } = io
```
Comment (niall-shaw) [inline src/context.ts:14]: Why not just pass the `database` directly? Do we expect more IO params in the future?

File: src/entity/mongoRepository.ts [hunk 2]
```diff
@@ -22,8 +22,10 @@ export async function createEntityRepository(
   return {
     getAllEntities: partial(getAllEntities, collection),
     findById: partial(findById, collection),
+    findByDid: partial(findByDid, collection),
     addEntity: partial(addEntity, collection),
     updateEntity: partial(updateEntity, collection),
+    deleteAll: partial(deleteAll, collection),
```
Comment (niall-shaw) [inline src/entity/mongoRepository.ts:28]: Should we have `deleteAll` in production?

Comment (niall-shaw) [reply src/entity/mongoRepository.ts:28]: It seems we are only using it for tests?

Comment (jakubkoci) [reply src/entity/mongoRepository.ts:28]: Fair point. I removed it from repository interfaces and now droppping the whole db before each test. The tests still run quite fast.

File: src/context.ts (hunk 1 above)
Comment (jakubkoci) [reply src/context.ts:14]: Yes. Honestly, I wasn't sure about this step but it seemed a bit weird to pass it just as one parameter. We might not need to add more deps like this in the near future but if we do we should not add it as another parameter but pass it as a property of one deps object... the naming `io` was just first idea that came to my mind, maybe not so great 🤷‍♂️ but it kind of make sense to me.

**Review by niall-shaw — APPROVED** [review]

**Review by niall-shaw — APPROVED** [review]

**Pull request merged**